serializers/src/main/java/io/pravega/schemaregistry/schemas/JSONSchema.java

```diff
@@ -0,0 +1,150 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.schemas;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchema;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchemaGenerator;
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableMap;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.contract.data.SerializationFormat;
+import lombok.Getter;
+import org.apache.avro.specific.SpecificRecordBase;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Container class for Json Schema.
+ *
+ * @param <T> Type of element. 
+ */
+public class JSONSchema<T> implements Schema<T> {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Getter
+    private final String schemaString;
+    private final Class<T> base;
+    @Getter
+    private final Class<? extends T> tClass;
+
+    @Getter
+    private final JsonSchema schema;
+
+    private final SchemaInfo schemaInfo;
+
+    private JSONSchema(JsonSchema schema, String name, String schemaString, Class<T> tClass) {
+        this(schema, name, schemaString, tClass, tClass);
+    }
+
+    private JSONSchema(JsonSchema schema, String name, String schemaString, Class<T> base, Class<? extends T> derived) {
+        String type = name != null ? name : schema.getId();
+        // Add empty name if the name is not supplied and cannot be extracted from the json schema id. 
+        type = type != null ? type : "";
+        this.schemaString = schemaString;
+        this.schemaInfo = new SchemaInfo(type, SerializationFormat.Json, getSchemaBytes(), ImmutableMap.of());
+        this.base = base;
+        this.tClass = derived;
+        this.schema = schema;
+    }
+
+    private JSONSchema(SchemaInfo schemaInfo, JsonSchema schema, String schemaString, Class<T> tClass) {
+        this.schemaString = schemaString;
+        this.schemaInfo = schemaInfo;
+        this.base = tClass;
+        this.tClass = tClass;
+        this.schema = schema;
+    }
+
+    /**
+     * Method to create a typed JSONSchema for the given class. It extracts the json schema from the class.
+     * For POJOs the schema is extracted using jacksons {@link JsonSchemaGenerator}. 
+     *
+     * @param tClass Class whose object's schema is used.
+     * @param <T> Type of the Java class. 
+     * @return {@link JSONSchema} with generic type T that extracts and captures the json schema. 
+     */
+    public static <T> JSONSchema<T> of(Class<T> tClass) {
+        try {
+            JsonSchemaGenerator schemaGen = new JsonSchemaGenerator(OBJECT_MAPPER);
+            JsonSchema schema = schemaGen.generateSchema(tClass);
```

Comment:
So if its java enum class it can be passed to this JSONSchema.of(EnumClass.class). 
JsonSchema for enum is a string with fixed values from the enum. 

Since you asked about enum, there are primitive data types too and arrays. 
We have a method `of(JsonSchema schema)` which can be used to express those without an encapsulating class.